src/multiformats/codecs/index.ts

```diff
@@ -0,0 +1,23 @@
+import { ByteView } from 'multiformats/codecs/interface'
+import { BeeSon } from '../../beeson'
+import { JsonValue } from '../../types'
+
+export const name = 'beeson'
+
+export const code = 0xfc
+
+/**
+ * @param {BeeSon<JsonValue>} node
+ * @returns {ByteView<Uint8Array>}
+ */
+export const encode = (node: BeeSon<JsonValue>): ByteView<Uint8Array> => {
+  return node.serialize()
+}
+
+/**
+ * @param {ByteView<Uint8Array>} dataBytes
+ * @returns {Uint8Array}
+ */
+export const decode = (dataBytes: Uint8Array): Promise<BeeSon<JsonValue>> => {
+  return BeeSon.deserialize(dataBytes)
```

Comment:
for this method you potentially need `storageLoader` hook function to resolve superBeeSon references

Reply:
not yet, let's talk about this. `storageLoader` in IPFS/Fileocoin land is managed by the protocol, either the old one or the new one call `GraphSync`. The paper I added, `Swarm DB - Router`, is what we would use to query `storageLoader`. But you need to explain your ideas so I can mashup yours and GraphSync.

Reply:
`storageLoader` is a hook function being used to resolve SuperBeeSon/SuperType References. If you do not pass here and the BeeSon object contains any superBeeSon element it will throw an error because it cannot resolve the reference for typeSpecification, thereby it cannot deserialize the object.

You can initialize a storage object with `Utils.createStorage()` that has a `storageLoader` method, but you have to use its `storageSaverSync` method to cache superBeeSon values that can be resolved later.